### PR TITLE
Add filtering before DeleteOrEvictPods

### PR DESCRIPTION
**Pull request by tiationg-kho**
**Issue #, if available:**
#1097

**Description of changes:**
- Add filtering before DeleteOrEvictPods

**How you tested your changes:**
Environment (Linux / Windows): Linux
Kubernetes Version: 1.31

Set `use-apiserver-cache: true` and `emit-kubernetes-events: true` then toggle the `ignore-daemon-sets` flag to test protect daemonset pods or not.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by LikithaVemulapalli — COMMENTED** [review]
quick question: Is there existing test to test the functionality locally either in test or internal-test? If test doesn't exist can we test this locally by having a mock setup?

**Comment by tiationg-kho** [comment]
> quick question: Is there existing test to test the functionality locally either in test or internal-test? If test doesn't exist can we test this locally by having a mock setup?

Add a unit test case for FilterOutDaemonSetPods method.

File: pkg/node/node_test.go [hunk 1]
```diff
@@ -377,3 +378,43 @@ func TestUncordonIfRebootedTimeParseFailure(t *testing.T) {
...
+					},
+				},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "mock-regular-pod",
+				},
+			},
+		},
+	}
```
Comment (LikithaVemulapalli) [inline pkg/node/node_test.go:416]: nit: indentation seems off

**Review by LikithaVemulapalli — APPROVED** [review]
/lgtm

**Pull request merged**